lite/backends/imagination_nna/imgdnn_manager.h

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <unistd.h>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "imgdnn.h"  // NOLINT
+#include "lite/core/op_lite.h"
+#include "lite/core/tensor.h"
+
+namespace paddle {
+namespace lite {
+namespace imagination_nna {
+
+class ImgdnnManager {
+  imgdnn_err_code err_;
+  imgdnn_device device_;
+  imgdnn_network net_{nullptr};
+  imgdnn_context context_{nullptr};
+  imgdnn_binding binding_{nullptr};
+  imgdnn_network_object net_obj_{nullptr};
+  std::vector<uint8_t *> coef_pool;
+
+  imgdnn_tensor convertQuantTensorType(imgdnn_tensor a_tensor,
+                                       imgdnn_quant_param *dst_quant_param);
+
+  bool testConfigFileExists(const std::string &hwconfig,
+                            const std::string &mapconfig);
+
+ public:
+  ImgdnnManager();
+
+  virtual ~ImgdnnManager() {
+    if (net_obj_) err_ = imgdnnNetworkObjectDestroy(net_obj_);
+    if (context_) err_ = imgdnnContextDestroy(context_);
+    if (binding_) err_ = imgdnnBindingDestroy(binding_);
+    if (net_) err_ = imgdnnNetworkDestroy(net_);
+
+    for (auto buf : coef_pool) delete[] buf;
+  }
+
+  uint8_t *GetBufromPool(size_t size) {
+    uint8_t *buf = new uint8_t[size];
```

Comment:
seems a std::vector<uint8_t> is better, no need to use a `new`?

avoid using a raw `new` or `delete`, wrap it with a smart pointer.

```c++
std::vector<uint8_t> pool;

uint8_t *GetBufromPool(size_t size) {
  pool.resize(pool.size()+size);
  return pool.data() + pool.size()-size;
}
```

Reply:
This is a good idea, and works in my small test program, but crash when integrate into paddle-lite tree. Using unique_ptr instead.